docs/stories/EPIC-Restore deleted messages/TF-1719-User-can-restore-deleted-messages.md

```diff
@@ -0,0 +1,57 @@
+# Summary
+
+* [Related EPIC](#related-epic)
+* [Definition](#definition)
+* [Screenshots](#screenshots)
+* [Misc](#misc)
+
+## Related EPIC
+https://github.com/linagora/tmail-flutter/issues/1719
+
+## Definition
+
+- Given that I am a Tmail user and I have logged in successfully
+- When I click on three-dot button of folder Trash, I can see a new option in context menu: Recover deleted messages 
+- I select this option, a popup will be opened:
+- In the popup, I can input fields:
+   - Deletion date: 
+      + Default option is "Last 1 year"
+      + I can select other options: Last 7 days, Last 30 days, Last 6 months, Custom range 
+      + When I select Custom range, a Date picker will be displayed that allow me to select From date and To date.
+      + The date range is limited between today to one year ago. 
+   - Reception date: 
+      + Default option is "All time"
+      + I can select other options: Last 7 days, Last 30 days, Last 6 months, Last year, Custom range 
+      + When I select Custom range, a Date picker will be displayed that allow me to select From date and To date.
+      + The date range is not limited 
+   - Subject: a text field 
+   - Recipients: I can input name/email address, the system will display a suggestion list of matched emails. I can select from the list or input the recipients mannually. I can input multiple recipients.
+   - Sender:  I can input name/email address, the system will display a suggestion list of matched emails. I can select from the list or input the sender mannually. I can input only 1 sender.
+   - Has attachment: a Checkbox. If it is checked, the system will find and restore email which contains attachments
+- After input conditions, I click button Restore, the system will search for matched messages in in deleted vault. 
+- When the recovery is in progress, there will be a progess bar with a message :"Recovery is in progress and might take time. Please wait a few minutes."  If I click Cancel button, the restoring process stops immediately. When I click on Close button, the popup is close but the recovery is still in progress. 
```

Comment:
What happens if user close the dialog and then want to cancel the recovery process?